impl/src/main/java/org/jboss/weld/event/ObserverNotifier.java

```diff
@@ -324,7 +330,7 @@ public RuntimeException apply(Type eventType) {
         }
         // We should always initialize and validate all notification options first
         final NotificationMode mode = initMode(options.get(WeldNotificationOptions.MODE));
-
+        final Long timeout = verifyNotificationTimeoutOption(options);
```

Comment:
We should unify the init/validation method names. If you don't like the `initX` template we could change it:
* `initModeOption()`/`initTimeoutOption()`
* `initializeMode()`/`initializeTimeout()`
* `verifyModeOption()`/`verifyTimeoutOption()`
* ...